### ISSUE TITLE
Move or retire UserInterface::eraseCredentials

### ISSUE THREAD

**Issue by dbrumann** [symfony/symfony#57842]
### Description

This comes a bit out of the blue and I am not sure how relevant it is or if it was asked before (at least I couldn't find any issues after a quick search). I am setting up a new demo project and having to implement an empty `UserInterface::eraseCredentials` irked me a bit.

Essentially, it seems misplaced to me after moving `getPassword` out and possibly even no longer relevant although that might be specific to how I use the security system. In any case, we have the same method `eraseCredentials` both on the `UserInterface` and `TokenInterface` and quite often - even in the examples in the docs nowadays - it is just kept empty. I suggest deprecating both and moving them to a new dedicated, optional interface. Before starting the hardest task of finding a name for that new interface I wanted to ask if it is even worth the effort.

Should `eraseCredentials` from `UserInterface` and `TokenInterface` be merged into its own dedicated interface?

**Comment by chalasr** [comment]
Sounds good to me. Good luck finding the name :)

**Comment by chalasr** [comment]
See #59106
